compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -631,7 +631,7 @@ private BType setLiteralValueAndGetType(BLangLiteral literalExpr, BType expType)
         } else if (literalType.tag == TypeTags.DECIMAL) {
             return decimalLiteral(literalValue, literalExpr, expType);
         } else if (literalType.tag == TypeTags.STRING && types.isCharLiteralValue((String) literalValue)) {
-            if (expType.tag == TypeTags.CHAR_STRING) {
+            if (expType.tag == TypeTags.CHAR_STRING || expType == symTable.noType) {
```

Comment:
Shouldn't the following still work?

```ballerina
public function main() {
    var s = "a";
    s = "hello";
}
```

Seems to result in an error (`incompatible types: expected 'string:Char', found 'string'`) with these changes.

Related section in the spec - https://ballerina.io/ballerina-spec/spec.html#section_6.2.3

Reply:
fixed